src/renderer/pages/provider.vue

```diff
@@ -205,13 +213,33 @@ export default {
         logger.warn(e)
       }
 
-      this.pendingRequest = false
+      this.pendingStopRequest = false
     },
 
     onStatusChange (newStatus) {
       this.status = newStatus
       // TODO: show error if status changes from "Starting" to "NotRunning"
       // TODO: show error if service ends unexpectedly, without stoping service
+    },
+    async stopAndGoToVpn () {
+      await this.providerService.stop()
+
+      this.goToVpn()
+    },
+    async onTabClick (page) {
+      if (page !== 'vpn') {
+        return
+      }
+
+      if (this.status !== ServiceStatus.NOT_RUNNING || this.pendingStartRequest) {
```

Comment:
?? Why are we not checking `pendingStopRequest`?
```suggestion
      if (this.status !== ServiceStatus.NOT_RUNNING || this.pendingStartRequest || this.pendingStopRequest) {
```